packages/nextclade-web/src/io/fetchSingleDatasetAuspice.ts

```diff
@@ -0,0 +1,38 @@
+import { isEmpty } from 'lodash'
+import { FatalError } from 'next/dist/lib/fatal-error'
+import { attrStrMaybe, AuspiceTree, Dataset, DatasetFiles } from 'src/types'
+import { removeTrailingSlash } from 'src/io/url'
+import { axiosFetch } from 'src/io/axiosFetch'
+
+export async function fetchSingleDatasetAuspice(datasetJsonUrl_: string) {
+  const datasetJsonUrl = removeTrailingSlash(datasetJsonUrl_)
+
+  const auspiceJson = await axiosFetch<AuspiceTree>(datasetJsonUrl)
+  const pathogen = auspiceJson.meta?.extensions?.nextclade?.pathogen
+
+  if (isEmpty(auspiceJson.root_sequence?.nuc)) {
+    throw new FatalError(`Auspice JSON does not contain required field '.root_sequence.nuc': ${datasetJsonUrl_}`)
+  }
+
+  const currentDataset: Dataset & { auspiceJson?: AuspiceTree } = {
+    path: datasetJsonUrl,
+    capabilities: {
+      primers: false,
+      qc: [],
+    },
+    ...pathogen,
+
+    // HACK: there is no files if dataset comes from Auspice JSON, neither they are needed. What to do?
+    files: {} as unknown as DatasetFiles,
```

Comment:
Just following up on this, @rneher's [example work on h5n1 (not yet merged)](https://github.com/nextstrain/avian-flu/pull/33/commits/f15bf37345fcc39fb8978b7bc87e3d589c5bf9aa) specified a `files` section here. Perhaps that was an experiment and you've decided to go in a different direction, but I agree with the code as written here - that `files` doesn't make sense for an Auspice JSON. Perhaps if they weren't relative file paths but were HTTP addresses then `examples` makes sense, but I wouldn't think it's worth it, and the contents of the other "files" are instead encoded in the Auspice dataset itself.

Reply:
Slight complication, or maybe my misunderstanding, in the above linked commit `files` are encoded via `auspiceJson.meta.extensions.nextclade.pathogen.attributes.files` so they are separate to the (empty) `files` in this line. Is one of them a typo, or are the different things?

Reply:
@ivan-aksamentov I wrote a little more about this in https://github.com/nextstrain/avian-flu/pull/33#issuecomment-2128273038

Reply:
@jameshadfield I made `files` object and all of it fields optional in [`ddd9925` (#1455)](https://github.com/nextstrain/nextclade/pull/1455/commits/ddd9925848796f4b33d174b8a42aefd40cd09660)

Reply:
Yes - working now! Thanks 🙏